### ISSUE TITLE
GraphQL InputType argument parsing error & workaround

### ISSUE THREAD

**Issue by patty-caked** [EntityGraphQL/EntityGraphQL#31]
The following gives me what seems to be a parsing error: 

> **_Error: line 2:45 no viable alternative at input '['_**

```
mutation BrokenNestedInput {
  nestedArgumentQuery(dryRun: true, filters: [{field: itemId, equals: "1234"}]) {
    itemId
    itemName
  }
}
```
As far as I can tell, this happens any time I try to pass an input type within an input type, and a similar error occurs with '{' when I change filters to only accept a single element.

I discovered a workaround where I pass filters the following query variable:
```
## Query Variables
{
  "filter": 
  [
    {
      "field": "itemId", 
      "equals": "1234"
    }, 
    {
      "field": "itemId",
      "equals": "4321",
      "conjunction": "OR"
    }
  ]
}
----------
##"Query"
mutation NestedInputTypeTest($filter: [FilterInput]) {
  nestedArgumentQuery(filters: $filter) {
    itemId
    itemName
  }
}
```
This gives me the results I want with no parsing errors.

Also, just for clarification, I'm using mutation methods as makeshift queries to test a filtering system.

**Comment by lukemurray** [comment]
Interesting. I'm not sure if the GQL spec actually supports JSON at an argument level. I'll have a look. If it doesn't this isn't something I'd like at implementing. If it does, then I will although it is lower priority as you can use the query variables, which is how the docs suggest doing it as well

**Comment by patty-caked** [comment]
All I know is that Wayfair's GraphQL portal for partners accepts it like that just fine, which is what I'm basing my own project on. Though you're right, it's not a high priority issue.
How difficult would it be to add support for this? I'm guessing it wouldn't be as "simple" as adding a few extra things to the EntityGraphQL.g4 file, right?

**Comment by lukemurray** [comment]
It’ll be adding the json spec to the grammar file and then support processing that in the parser code


________________________________

**Comment by lukemurray** [comment]
Fix coming in next release

**Comment by lukemurray** [comment]
Fixed in 1.0.x
